### ISSUE TITLE
Request interceptor - handling csrf token

### ISSUE THREAD

**Issue by MateuszPe** [mrin9/RapiDoc#435]
Hi! 

In our application we have to handle csrf token by double submit cookie pattern and for now I am not sure how to achieve this.
Is there a possibility in Rapidoc to use request interceptor to manipulate headers? 

Thanks!

**Comment by mrin9** [comment]
we dont have one, but we have somewhere in our roadmap to provide one.
But, can you provide me with little more details on your requirements. 
- Are you trying to submit `cookie` or add a `request-header` ?
- do you see a need to ever include a cookie that must be sent to `cross-origin` / `same-origin` ?

will help us document our requirements

**Comment by MateuszPe** [comment]
> Are you trying to submit cookie or add a request-header ?

I'm trying to add `request-header`.  

> do you see a need to ever include a cookie that must be sent to cross-origin / same-origin

No, I don't see such need..

What I would like to achieve is to extract the `csrf` cookie value before each request and add its value to `X-XSRF-TOKEN` request-header.


![csrf (1)](https://user-images.githubusercontent.com/25302444/110864680-bb259200-82c2-11eb-97ce-423d37499a3d.jpg)

**Comment by mrin9** [comment]
thanks for the explanation and the image.
I am guessing the `csrf_id` you receive is very short lived ?

If its valid for little long duration (lets say abt 2 hrs)
then there is a `spec-loaded` event in RapiDoc on which you can make a call to `POST /login` and get the `csrf_id`
after this you can update following 3 attribute of RapiDocs
 -  `api-key-name='X-XSRF-TOKEN'`
 -  `api-key-location='header'`
 -  `api-key-value={value you receive from POST /login} `

after this RapiDoc will insert this header in every TRY call  

for an example have a look at https://github.com/mrin9/RapiDoc/blob/master/docs/examples/example8.html on how to supply your own header/value

**Comment by MateuszPe** [comment]
I made mistake on previous image. In the picture below, I highlighted in red the part that I previously missed.
![csrf (1)](https://user-images.githubusercontent.com/25302444/111084259-d508e400-8511-11eb-8447-7952b6c6f300.png)

I cannot use proposed approach. `csrf_if` may be short lived.
It would be nice to have similiar functionality to `requestInterceptor` and `responseInterceptor` provided in [SwaggerUI](https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/).
